From f64c3d9b648cdc8d48a9aea475f4bad28d863518 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 18 Aug 2023 08:18:58 +0300
Subject: [PATCH] QtVFS: fix build with sqlite version < 3.40

sqlite3_filename typedef was introduced in sqlite v3.40 (2022-11-16).
Add typedef to work with older versions.

Change-Id: I80b5ed42c03eb962a86bbab53591c84e006d2c44
Upstream-Status: Submitted
---
 src/plugins/sqldrivers/sqlite/qsql_sqlite_vfs.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/plugins/sqldrivers/sqlite/qsql_sqlite_vfs.cpp b/src/plugins/sqldrivers/sqlite/qsql_sqlite_vfs.cpp
index e6e5789939..c5bf720b34 100644
--- a/src/plugins/sqldrivers/sqlite/qsql_sqlite_vfs.cpp
+++ b/src/plugins/sqldrivers/sqlite/qsql_sqlite_vfs.cpp
@@ -9,6 +9,9 @@
 #include <sqlite3.h>
 #include <stdio.h> // defines FILENAME_MAX everywhere
 
+// needed for sqlite < 3.40
+typedef const char *sqlite3_filename;
+
 #ifndef PATH_MAX
 #  define PATH_MAX FILENAME_MAX
 #endif
